sorting/cycle_sort.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * @file
+ * @brief Implementation of [Cycle sort](https://en.wikipedia.org/wiki/Cycle_sort) algorithm
+ *
+ * @details
+ * Cycle Sort is a sorting algorithm that works in \f$O(n^2)\f$ time in best cas and works in \f$O(n^2)\f$ in worst case.
+ * If a element is already at its correct  position, do nothing.
+ * If a element is not at its correct position, we then need to move it to its correct position by computing the correct positions.Therefore, we should make sure the duplicate elements.
+ *
+ * @author [TsungHan Ho](https://github.com/dalaoqi)
+ */
+
+#include <algorithm> /// for std::is_sorted
+#include <cassert>     /// for assert
+#include <iostream> /// for io operations
+#include <vector>    /// for std::vector 
+
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+/**
+ * @namespace cycle_sort
+ * @brief Functions for [Cycle sort](https://en.wikipedia.org/wiki/Cycle_sort) algorithm
+ */
+namespace cycle_sort { 
+/**
+ * @brief Interchange a and b
+ * @tparam T type of entity
+ * @param a the first entity
+ * @param b the second entity
+ * @returns void
+ */
+template <typename T>
+void swap(T *a, T *b) {
+    T tmp = 0;
+    tmp = *a;
+    *a = *b;
+    *b = tmp;
+}
+
+/**
+ * @brief The main function implements cycleSort
+ * @tparam T type of array
+ * @param arr array to be sorted
+ * @returns void
+ */
+template <typename T>
+void cycleSort(std::vector<T> *arr) {
```

Comment:
a better approach might be as below. This is only a suggestion and is up to you if you want to implement.
```suggestion
std::vector<T> cycleSort(const std::vector<T> &in_arr) {
    std::vector<T> arr(in_arr); // create a copy of the input array, process it and return a new array. Thus, the input array does not get modified and no need for pointers :)
```